### PR TITLE
feat(addon-editor): add `highlight` extension

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/addon-editor/extensions/highlight/highlight.extension.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                return this.editor.commands.insertContent(`\t`);
+            },
+
+            Enter: () => this.editor.commands.insertContent(`<p></p>`),
+        };
+    },
+}).configure({
+    multicolor: true,
+    HTMLAttributes: {
+        class: `marked`,
```
Comment (waterplea) [inline projects/addon-editor/extensions/highlight/highlight.extension.ts:38]: Do we need this class? Maybe people can just use `mark` tag name?

File: projects/addon-editor/extensions/highlight/highlight.extension.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import {getHTMLFromFragment} from '@tiptap/core';
+import {Highlight} from '@tiptap/extension-highlight';
+
+export const TuiHighlight = Highlight.extend({
+    addKeyboardShortcuts(): any {
+        return {
+            Tab: () => {
+                if (this.editor.isActive(`highlight`)) {
```
Comment (waterplea) [inline projects/addon-editor/extensions/highlight/highlight.extension.ts:8]: Can you explain this code a little bit? Why `Tab` and `Enter` have some specific actions inside `mark`?

File: projects/addon-editor/extensions/highlight/highlight.extension.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-editor/extensions/highlight/highlight.extension.ts]: fixed

File: projects/addon-editor/extensions/highlight/highlight.extension.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-editor/extensions/highlight/highlight.extension.ts]: fixed

File: projects/demo/src/modules/components/editor/highlight/editor-highlight.module.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import {ExampleTuiEditorHighlightComponent} from './editor-highlight.component';
+import {TuiEditorHighlightExample1} from './examples/1';
+
+@NgModule({
+    imports: [
+        CommonModule,
+        TuiActiveZoneModule,
+        TuiSvgModule,
+        TuiLinkModule,
+        TuiHostedDropdownModule,
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/editor/highlight/editor-highlight.module.ts:24]: Does this example actually need all these imports ?

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/components/editor/highlight/editor-highlight.module.ts (hunk 3 above)
Comment (splincode) [reply projects/demo/src/modules/components/editor/highlight/editor-highlight.module.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**